src/main/java/hudson/plugins/ec2/CloudHelper.java

```diff
@@ -66,4 +73,39 @@ static Instance getInstance(String instanceId, EC2Cloud cloud) throws AmazonClie
         }
         return instances.get(0);
     }
+
+    @CheckForNull
+    static Image getAmiImage(AmazonEC2 ec2, String ami) {
+        List<String> images = new LinkedList<>();
```

Comment:
You could maybe use Collections.singleton() or empty() to simplify this (even if you only moved this).

Reply:
@varyvol This would be better? (genuine question, my Java skills are rusty)

```java
        List<String> images = Collections.emptyList();
        images.add(ami);
        List<String> owners = Collections.emptyList();
        List<String> users = Collections.emptyList();
```

Reply:
@multani something like this:

```suggestion
        List<String> images = Collections.singletonList(ami);
```

Reply:
Precisely that, including @timja 's suggestion.